### PR TITLE
Add coverage to codacy

### PR DESCRIPTION
**Pull request by machadoit**
Add badges to docs.
Add coverage to codacy.
Add some tests.
Changed some strings, and some extract method.

### CODE REVIEW

File: src/main/scala/com/codacy/CodacyCoverageReporter.scala [hunk 1]
```diff
@@ -111,8 +115,11 @@ object CodacyCoverageReporter {
                 Right(s"Coverage data uploaded. ${requestResponse.message}")
             }
         }).joinRight
+    }
+  }
 
-    } match {
+  def codacyCoverage(config: Config): Unit = {
+    coverageWithTokenAndCommit(config) match {
```
Comment (mrfyda) [inline src/main/scala/com/codacy/CodacyCoverageReporter.scala]: Codacy Issue found: [Default case not specified.](https://www.codacy.com/app/Codacy/codacy-coverage-reporter/file/1269767061/issues/source?bid=2276540&fileBranchId=2942038#l122)

Comment (mrfyda) [inline src/main/scala/com/codacy/CodacyCoverageReporter.scala]: Codacy Issue found: [Default case not specified.](https://www.codacy.com/app/Codacy/codacy-coverage-reporter/file/1299817504/issues/source?bid=2276540&fileBranchId=2942038#l122)

File: src/main/scala/com/codacy/CodacyCoverageReporter.scala [hunk 2]
```diff
@@ -43,17 +43,35 @@ object CodacyCoverageReporter {
 
   def main(args: Array[String]): Unit = {
 
-    val parser = new scopt.OptionParser[Config]("codacy-coverage-reporter") {
+    val parser = buildParser
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/CodacyCoverageReporter.scala]: Why are we renaming this value?

File: src/main/scala/com/codacy/CodacyCoverageReporter.scala [hunk 3]
```diff
@@ -43,17 +43,35 @@ object CodacyCoverageReporter {
...
+
+        codacyCoverage(config)
+      case Some(config) if config.projectToken.trim.isEmpty =>
+        logger.error("Error: Missing option --projectToken")
+      case _ =>
+    }
+
+  }
+
+  def buildParser = {
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/CodacyCoverageReporter.scala]: If there are not too much dependencies extract this method form main

Comment (machadoit) [inline src/main/scala/com/codacy/CodacyCoverageReporter.scala]: It is out of main, github mind tricks!

**Pull request merged**